### DISCUSSION TITLE
Possibility to use placeholders in automations

### DISCUSSION THREAD

**Discussion by slungs** [home-assistant/frontend#21281] [Automations / scripts / scenes]
As a user I would like to be able to use placeholders in automations, so that I can execute an idea for an automation even though I don't have all devices necessary for it to function at the moment.

For example: I want to create an automation for the house to go into "night mode" when I plug my charger in at my bedside at night. For that, I need to add a wall plug to my charger - but since I am not at home right now, I can not create the automation right now. It would be neat to be able to put a placeholder for this device/entity in the automation, so that I can build out the automation right now when the idea is in my head, and then finish it later. 

There could also be some kind of "draft" state in the overview, similar to the disabled/unavailable state in the ui, to signal this.

**Comment by Misiu** [comment]
Use a template that you will update with real device. For example, Shelly mini EM shows you power usage. I use this template to know if there is something plugged in to socket:
```
{{states('sensor.garage_socket_power') | float > 0}}
```
I use this in a binary sensor template.
This way when your automation works, you just update the template.

**Reply by slungs** [reply]
> Use a template that you will update with real device. For example, Shelly mini EM shows you power usage. I use this template to know if there is something plugged in to socket:
> 
> ```
> {{states('sensor.garage_socket_power') | float > 0}}
> ```
> 
> I use this in a binary sensor template. This way when your automation works, you just update the template.

Yes, there are of course workarounds. But hardly the most user friendly.
